src/visualizer/visualizer.js

```diff
@@ -16,11 +16,25 @@ const week = "week";
 const month = "month";
 const year = "year";
 
+const getMaxPriorityElement = prioritized => {
+    const max_priority = [prioritized[0]]
+    prioritized.slice(1).forEach((item) => {
+        if (item.priority === max_priority[0].priority) {
+            max_priority.push(item)
+        }
+    })
+
+    var index = Math.floor((Math.random() - 0.001) * max_priority.length)
+    return max_priority[index]
+}
+
 const chooseRandomDeadline = deadlines => {
-    const random = Math.ceil(Math.random() * 100);
-    const index = random % deadlines.length;
+    // first sort the array based on priority
+    const prioritized = deadlines.sort((a, b) => (b.priority - a.priority))
+    // check if the priority elements have multiple items
+    const max_priority = prioritized.length === 1 ? prioritized[0] : getMaxPriorityElement(prioritized)
```

Comment:
If there are more than 1 deadline with all of them have no priority, this function `getMaxPriorityElement(prioritized)` will always return the first index. I think we still need to randomize the deadline when there's no priority on all existing deadlines

Reply:
Each element will have priority. There is no `No priority` case. If the user does not set the priority then it will be set to `0.5` by default.